src/sentry/api/endpoints/release_file_details.py

```diff
@@ -104,6 +116,9 @@ def get(self, request, project, version, file_id):
         except ReleaseFile.DoesNotExist:
             raise ResourceDoesNotExist
 
+        if request.GET.get('download') is not None and (
+           request.access.has_scope('project:write')):
+            return self.download(releasefile)
```

Comment:
We should also probably return a 403 here when requesting a download but don't have have permission.